opentech/apply/projects/models.py

```diff
@@ -73,14 +81,67 @@ def get_remove_form(self):
         return RemoveDocumentForm(instance=self)
 
 
+class PaymentApproval(models.Model):
+    request = models.ForeignKey('PaymentRequest', on_delete=models.CASCADE, related_name="approvals")
+    by = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE, related_name="payment_approvals")
+
+    created_at = models.DateTimeField(auto_now_add=True)
+
+    def __str__(self):
+        return f'Approval for {self.request} by {self.by}'
+
+
+class PaymentReceipt(models.Model):
+    payment_request = models.ForeignKey("PaymentRequest", on_delete=models.CASCADE, related_name="receipts")
+
+    file = models.FileField(upload_to=receipt_path)
+
+    def __str__(self):
+        return f'Receipt for {self.payment_request}'
+
+
+SUBMITTED = 'submitted'
+UNDER_REVIEW = 'under_review'
+PAID = 'paid'
+DECLINED = 'declined'
+REQUEST_STATUS_CHOICES = [
+    (SUBMITTED, 'Submitted'),
+    (UNDER_REVIEW, 'Under Review'),
+    (PAID, 'Paid'),
+    (DECLINED, 'Declined'),
+]
+
+
+class PaymentRequest(models.Model):
+    project = models.ForeignKey("Project", on_delete=models.CASCADE, related_name="payment_requests")
+    by = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE, related_name="payment_requests")
+
+    invoice = models.FileField(upload_to=invoice_path)
```

Comment:
Needs to upload to the private storage as per #1417 - i think we need to merge sprint-1 into 2 to allow that.